extra_geom/detectors.py

```diff
@@ -854,6 +854,170 @@ def to_distortion_array(self, allow_negative_xy=False):
+        detector. Coordinates increase upwards and to the left (looking along
+        the beam).
+
+        To give positions in units other than pixels, pass the *unit* parameter
+        as the length of the unit in metres. E.g. ``unit=1e-3`` means the
+        coordinates are in millimetres.
+        """
+        asic_gap_px = asic_gap * unit / cls.pixel_size
+        panel_gap_x = panel_gap[0] * cls.pixel_size
+        panel_gap_y = panel_gap[1] * cls.pixel_size
+
+        # How much space one tile takes up, including the gaps
+        # separating it from its neighbour.
+        # In the y dimension, 128 px
+        module_height = cls.frag_fs_pixels * cls.pixel_size
+        # In x, 64 px + gap between tiles (asics)
+        tile_width = (cls.frag_ss_pixels + asic_gap_px) * cls.pixel_size
+        module_width = 8 * tile_width
+
+        modules = []
+        for p in range(cls.n_modules):
+            panel_corner_y = (center[1] * unit) - ((p // 2) * (module_height + panel_gap_y))
+            panel_corner_x = (center[0] * unit) + ((p % 2) * (module_width + panel_gap_x))
+
+            tiles = []
+            modules.append(tiles)
+
+            for a in range(cls.n_tiles_per_module):
+                corner_x = panel_corner_x + a * tile_width
+
+                tiles.append(GeometryFragment(
+                    corner_pos=np.array([corner_x, panel_corner_y, 0.]),
+                    ss_vec=np.array([1, 0, 0]) * unit,
+                    fs_vec=np.array([0, 1, 0]) * unit,
+                    ss_pixels=cls.frag_ss_pixels,
+                    fs_pixels=cls.frag_fs_pixels,
+                ))
+        return cls(modules)
+
+    def inspect(self, axis_units='px', frontview=True):
+        """Plot the 2D layout of this detector geometry.
+
+        Returns a matplotlib Axes object.
+
+        Parameters
+        ----------
+
+        axis_units : str
+          Show the detector scale in pixels ('px') or metres ('m').
+        frontview : bool
+          If True (the default), x increases to the left, as if you were looking
+          along the beam. False gives a 'looking into the beam' view.
+        """
+        ax = super().inspect(axis_units=axis_units, frontview=frontview)
+        scale = self._get_plot_scale_factor(axis_units)
+
+        # Label modules and tiles
+        for ch, module in enumerate(self.modules):
+            cx, cy, _ = module[4].centre() * scale
+            ax.text(cx, cy, f'Mod{ch}', fontweight='bold',
+                    verticalalignment='center',
+                    horizontalalignment='center')
+
+            for t in [0, 7]:
+                cx, cy, _ = module[t].centre() * scale
+                ax.text(cx, cy, 'T{}'.format(t + 1),
+                        verticalalignment='center',
+                        horizontalalignment='center')
+
+        ax.set_title(f'AGIPD-500K2G detector geometry ({self.filename})')
+        return ax
+
+    def _get_dimensions(self):
```

Comment:
I don't think we actually need this unless we define a method to assemble data by interpolation - we did that for AGIPD-1M, but it's horrendously slow (maybe there's a better way to do it that I didn't find).

Reply:
I added the `position_modules_interpolate` method